rpmlb/builder/base.py

```diff
@@ -94,27 +95,40 @@ def build_with_retrying(self, package_dict, **kwargs):
     def build(self, package_dict, **kwargs):
         raise NotImplementedError('Implement this method.')
 
-    def edit_spec_file(self, spec_file):
-        spec_file_origin = '{0}.orig'.format(spec_file)
-        os.rename(spec_file, spec_file_origin)
-        fh_r = None
-        fh_w = None
-        try:
-            fh_r = open(spec_file_origin, 'r')
-            fh_w = open(spec_file, 'w')
-            fh_w.write('# Edited by rpmlb\n')
-            yield(fh_r, fh_w)
-        finally:
-            if fh_w:
-                fh_w.close()
-            if fh_r:
-                fh_r.close()
+    @staticmethod
+    @contextmanager
+    def edit_spec_file(target: Path):
+        """Safely edit a SPEC file in-place.
+
+        The target is backed up as '{target}.orig' if needed.
+
+        Keyword arguments:
+            target: The modified SPEC file path.
+
+        Returns:
+            Context manager providing open handles
+            for input and output file.
+        """
+
+        # Ensure path type
+        target = Path(target) if not isinstance(target, Path) else target
+
+        # Back up the original
+        source = target.with_suffix('.spec.orig')
+        if not source.exists():
+            target.rename(source)
+
+        # Provide the handles
+        with source.open(mode='r') as src, target.open(mode='w') as tgt:
```

Comment:
> Maybe we can discuss the 80-character-per-line limit in separate issue – personally, I have it set to 100 and have far less issues as this.

OK, please create new ticket with the reason why you prefer 100 than 80 to discuss that.
As a information, `flake8`'s default max length for a line is 79.